src/lib/y2packager/medium_type.rb

```diff
@@ -0,0 +1,107 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2019 SUSE LLC, All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# ------------------------------------------------------------------------------
+
+require "yast"
+
+require "y2packager/product_location"
+require "y2packager/repomd_downloader"
+
+Yast.import "InstURL"
+
+module Y2Packager
+  # This class scans the installation medium type and detects which installation
+  # medium type it is (online/offline/standard).
+  class MediumType
+    extend Yast::Logger
+
+    # Type of the installation medium, raises an exception if the installation
+    # URL is not set (nil) or is empty.
+    #
+    # @return [Symbol] Symbol describing the medium, one of `:offline`,
+    # `:online` or `:standard`
+    def self.type
+      @type ||= detect_medium_type
```

Comment:
hmm, this is quite confusing. What about returning instance of medium type, which has methods like `online?` and also name of this method I expect something like `detect` or `read` so it detect type.

Reply:
I'm not sure. I do not like the `detect` name because the detection is done only once and then the result is cached, this might be confusing. And I also want to avoid a long method chain like `Y2Packager::MediumType.detect.online?`, that's why there is the `Y2Packager::MediumType.online?` shortcut.

I introduced the `type` method just to make it easy to use it in `switch` although I expect that the most of the code will use just the `online?`/`offline?` methods.

Reply:
well, it can be better in situation when you want to store medium type like
```
medium_type = Y2Packager::MediumType.detect
if medium_type.online?
```

it also allows introspect and allows in future smarter methods in medium type, like "slow?` or `need_additional_ram?`

but it is up to you.

Reply:
OK, I see your point now. But it's more about the medium layout than about the physical properties of the medium. In this case we need to know if the medium provides packages for full installation (or at least for some minimal system) or it's just a boot medium and the packages are downloaded from SCC.

I'd keep it simple.

Thanks!